src/features/all-drops/components/AllDrops.tsx

```diff
@@ -157,56 +157,46 @@ export default function AllDrops() {
 
     setData(
       await Promise.all(
-        drops.map(
-          async ({
-            drop_id: id,
-            simple,
-            ft,
-            nft,
-            fc,
-            metadata = JSON.stringify({ dropName: 'untitled' }),
-            next_key_id,
-          }) => {
-            const meta = JSON.parse(metadata) || {};
-            if (!meta.dropName) {
-              meta.dropName = 'Untitled Drop';
-            }
+        drops.map(async ({ drop_id: id, simple, ft, nft, fc, metadata, next_key_id }) => {
+          const meta = JSON.parse(metadata || '{}');
```

Comment:
can we use getDropMetadata from keypom.ts? add the logic for defaulting the dropname to Untitled in the class as well. Easier to manage all logic if there are no duplication.

Reply:
done [here](https://github.com/keypom/keypom-airfoil/pull/150/commits/29ad7b2d5e781b446d3b20869d8b239a10a0e1be)

Reply:
sir, i think u missed out All Drops page